apex/contrib/csrc/groupbn/batch_norm_add_relu.h

```diff
@@ -399,7 +528,11 @@ class NhwcBatchNormAddRelu {
   // Calculate the expected fwd kernel occupancy, as dictated by shared memory usage.
   static int smem_driven_fwd_occupancy(int device_id, const int max_cta_per_sm) {
     using namespace at::cuda::utils;
+#ifdef __HIP_PLATFORM_HCC__
+    int fwd_reduction_bytes = THREADS_PER_PIXEL*(THREADS_PER_CTA/64)*ELEMENTS_PER_LDG*sizeof(float);
+#else
     int fwd_reduction_bytes = THREADS_PER_PIXEL*(THREADS_PER_CTA/32)*ELEMENTS_PER_LDG*sizeof(float);
```

Comment:
@sarunyap If this is due to the warp_size difference between Nvidia and AMD hardware, we should define a WARP_SIZE in a common header and use it everywhere we need warp-aware code, instead of hardcoding it this way.

Reply:
Yes, this is due to the different warp sizes.  Does APEX or PyTorch already have warp size defined somewhere that I can use?

Reply:
PyTorch defines the macro C10_WARP_SIZE in c10/macros/Macros.h.  But hip_runtime.h defines `warpSize` as a constexpr.  It depends on if we want the code to be buildable for both AMD and NV.  `warpSize` is only defined for device code on NV and is not a constexpr so cannot be used for fixed-size arrays.

Reply:
Thanks, Jeff.  So, I think it should be safe to use C10_WARP_SIZE?

Reply:
Yes, you should be able to use C10_WARP_SIZE.  c10/macros/Macros.h is an installed header for pytorch.

Reply:
@micmelesse @jithunnair-amd @jeffdaily I updated the code to use C10_WARP_SIZE